test/connector/tcp/mssql/test

```diff
@@ -1,17 +1,133 @@
-#!/bin/bash
+#!/bin/bash -e
 
-export SECRETLESS_HOST=secretless
+readonly db_host_tls="mssql"
+readonly db_host_no_tls="mssql" # TODO: configure a non-ssl container?
+readonly db_port=1433
+readonly db_user="sa"
+readonly db_password="yourStrong()Password"
+readonly secretless_port=2223
+secretless_host="secretless"
+benchmark=false
+
+help_and_exit() {
+  local retval=${1:-1}
+  cat <<EOF
+This script tests access to an MSSQL server, first accessing the
+server directly (to establish a testing baseline), and then
+via a secretless broker.
+
+This script detects if you're in "dev" mode by the existence of
+the secretless-dev container.  That container will exist only if
+the development process was started using the "./dev" command.
+
+Options:
+  -b runs benchmarks
+EOF
+  exit "$retval"
+}
+
+while getopts :b opt; do
+    case $opt in
+        b) benchmark=true;;
+       \?) echo "Unknown option -$OPTARG"; help_and_exit 1;;
+    esac
+done
+
+assert_contains() {
+  # assert_contains <string> <substring>
+  local string=$1
+  local substring=$2
+
+  if [[ "${string}" =~ ${substring} ]]; then
+      echo "."
+      return
+  fi
+
+  echo ""
+  echo "FAIL:"
+  echo ""
+  echo "The string:"
+  echo ""
+  echo "    ${string}"
+  echo ""
+  echo "Did not contain expected substring:"
+  echo ""
+  echo "    ${substring}"
+  echo ""
+
+  exit 1
+}
+
+query_sql_service() {
+  # query_sql_service <server> <port> <user> <password>
+  server=$1
+  port=$2
+  user=$3
+  pw=$4
+
+  docker_cmd="docker-compose run --rm --no-deps"
+  sql_cmd=(sqlcmd -S "$server,$port" -U "$user" -P "$pw" -Q "SELECT 1+1")
+
+  # NOTE: We need to 2>&1 because we need to assert on sqlcmd's error messages
+  # and we need to "|| true" because we don't want the script to exit when this
+  # happens
+  $docker_cmd test "${sql_cmd[@]}" 2>&1 || true
+}
+
+# Automatically detect if we're devmode based on the existence
+# of the secretless-dev container.  We assume that you started
+# your workflow using `./dev` if you are developing, and this
+# command will use the secretless-dev container.
 if [[ ! -z $(docker-compose ps -q secretless-dev) ]]; then
-  export SECRETLESS_HOST=secretless-dev
+  secretless_host="secretless-dev"
 fi
 
-docker-compose run \
-  -e SECRETLESS_HOST="$SECRETLESS_HOST" \
-  --rm \
-  --no-deps \
-  test bash -c "go test -v ./test/connector/tcp/mssql"
+echo "Waiting for '$secretless_host' service to start"
+
+# single quotes are intentional:
+# shellcheck disable=SC2016
+docker-compose run --rm --no-deps test bash -c '
+counter=0
+while ! wget --quiet --output-document - http://'$secretless_host':5335/ready > /dev/null 2>&1; do
+    if expr $counter = 5 > /dev/null; then
+      echo ""
+      echo "Timed out waiting for Secretless"
+      exit 1
+    fi
+    counter=$counter+1
```

Comment:
I don't think the code here does what you want it to do:
```
$ foo=0
$ foo=$foo+1
$ echo $foo
0+1
```

Reply:
```sh
let "counter = $counter + 1"
```
will work or you can use an arithmetic expansion `$(( ... ))`

Reply:
Yikes! Good catch!